tests/Integration/ClientTest.php

```diff
@@ -4,58 +4,58 @@
 
 namespace Werkspot\KvkApi\Tests\Integration;
 
-use GuzzleHttp\Client;
 use PHPUnit\Framework\TestCase;
-use Werkspot\KvkApi\Api\Profile\Company;
-use Werkspot\KvkApi\Client\Adapter\Guzzle;
-use Werkspot\KvkApi\Client\Adapter\Guzzle\Exception\NotFoundException;
-use Werkspot\KvkApi\Client\Authentication;
-use Werkspot\KvkApi\Client\Endpoint;
-use Werkspot\KvkApi\Client\Search\ProfileQuery;
-use Werkspot\KvkApi\ClientFactory;
+use Werkspot\KvkApi\Client\Profile\Company;
+use Werkspot\KvkApi\Http\Search\ProfileQuery;
+use Werkspot\KvkApi\Http\Endpoint\Testing;
+use Werkspot\KvkApi\KvkClientFactory;
 
 /**
  * @large
  */
 final class ClientTest extends TestCase
 {
-    private const USERNAME = 'testourapis';
-    private const PASSWORD = 'testourapis';
-
     /**
      * @test
      * @dataProvider getKvkNumbers
      */
     public function getProfile(int $kvkNumber): void
     {
-        $client = ClientFactory::getClient($this->getAdapter());
+        $client = KvkClientFactory::create('', new Testing());
 
         $profileQuery = new ProfileQuery();
         $profileQuery->setKvkNumber($kvkNumber);
 
-        try {
-            $profileResponse = $client->getProfile($profileQuery);
-        } catch (NotFoundException $e) {
-            self::assertTrue(true);
-
-            return;
-        }
+        $profileResponse = $client->getProfile($profileQuery);
 
-        foreach ($profileResponse->getCompanies() as $company) {
+        foreach ($profileResponse->getItems() as $company) {
             self::assertInstanceOf(Company::class, $company);
             self::assertEquals($kvkNumber, $company->getKvkNumber());
         }
     }
 
+    /**
+     * @test
+     * @dataProvider getNonExistingKvkNumbers
+     * @expectedException \Werkspot\KvkApi\Http\Adapter\Guzzle\Exception\NotFoundException
+     */
+    public function getProfile_shouldThrowException(int $kvkNumber): void
```

Comment:
maybe `shouldThrowExceptionWhenItemsAreNotFound`?

Reply:
I this case I think having a more explicit way is better since we are moving away from the happy flow.

Reply:
I see what you mean. I usually would not rely on the dataProvider to be explicit, the first place I look is at the method name.